src/dotty/tools/dotc/core/ConstraintHandling.scala

```diff
@@ -293,7 +352,13 @@ trait ConstraintHandling {
         case bound: PolyParam if constraint contains bound =>
           addParamBound(bound)
         case _ =>
-          val pbound = prune(bound)
+          var pbound = prune(bound)
+          if (pbound.isHK && !param.isHK) {
+            param match {
```

Comment:
Should this be `pbound match` ? Otherwise I don't get what's going on here, a comment would help